### ISSUE TITLE
Profile extension restart has no effect

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#142066]
From @bobbrow

I'm trying to profile the CMake Tools extension, but I can't get past the part where VS Code wants to restart.  
I run the "Show Running Extensions" command, right click on the extension I want to profile, then choose "start extension host profile". 

1. Popup -> restart
2. VS Code closes, but doesn't restart. 
3.  When I open the workspace again, profiling is not enabled and I get this message again :bug:

![image](https://user-images.githubusercontent.com/1926584/152322506-d25ed256-b75d-42f6-a78a-417ea139e1d0.png)

**Comment by jrieken** [comment]
Yes, that's not supported because the process disappears. You can try with `--prof-startup` in scenario that activates the extension early

**Comment by bobbrow** [comment]
I believe I tried `--prof-startup` but it looked like that just profiled the session startup time unless I misunderstood what was going on.  It wrote a couple of files after startup.

I want to profile the extension as it executes specific commands.  There was another option which looked promising, but I had no idea what to pass for `<port>`.  I didn't see any obvious URI's in the developer tools:
```
  --inspect-extensions <port>        Allow debugging and profiling of
                                     extensions. Check the developer tools for
                                     the connection URI.
```

**Comment by bobbrow** [comment]
I'm not looking to profile my extension's startup performance right now.  I want to profile it when it's doing certain tasks (post-activation).  @isidorn gave me some info offline about how to try doing that.  Is this feature in the Running Extensions tab only intended to provide startup performance profiling?

**Comment by jrieken** [comment]
Oh, sorry I misread that. I though you want to profile your extension during extension host restart... 

We don't always run the extension host in debug mode. When a user wants to profile we send it a signal to start to listen for debugging/profiling connections. When that fails we show the prompt which will restarting the aforementioned `--inspect-extensions` port. The editor should then automatically pick up the port and enable profiling. It seems that fails for you. Unsure why that is? Did you check the dev tools console from F1 > Show Developer Tools > Console? Do you have any firewall settings that would prevent the ext host from listening to a port? Is this windows? Iff so can you try on linux?

**Comment by bobbrow** [comment]
This is on Windows.  On linux the profiling works without a restart, but unfortunately what I'm trying to profile is why the code is slow on Windows, but not linux.  I get the same behavior on Windows for both Stable and Insiders.

I'm not sure what I'm looking for in the developer tools console or at what stage in the process you want me to inspect it.  When I click the restart button, VS Code shuts down, so I don't get an opportunity to see anything until I restart it manually myself and there's nothing significant in the console when I launch it myself.  I don't think I have any crazy firewall settings.

**Comment by jrieken** [comment]
Sorry for the long delay here... I do see the first ask to restart but then profiling works. Can you check with our process explorer (F1 > Open Process Explorer) and check if the extension host process has been launched with the `--inspect` flag (its `--inspect=6009` in the sample below)

<img width="1362" alt="Screen Shot 2022-02-22 at 11 35 27" src="https://user-images.githubusercontent.com/1794099/155114998-cd817e15-9243-4189-bc0e-cf37ec97f41b.png">

**Comment by bobbrow** [comment]
Sorry, now it's my turn to be delayed.  Unfortunately, I've been pre-empted by other tasks so it may be a while before I get back to this.

**Comment by jrieken** [comment]
ping @bobbrow

**Comment by bobbrow** [comment]
Go ahead and close this for now.  I'm not sure when I'll have time to get back to it.  Thanks!

**Comment by jrieken** [comment]
Ok
